Making `README` more user friendly

lidefi87 opened the pull request:
Reorganising `README` to make it more accessible to first time users as per [Issue #667](https://github.com/aodn/rimrep-dms/issues/667).

Issues encountered so far:
- I do not have access to create a GitHub pages from AODN repository. Any ideas of who can do this? Otherwise, we could use `bookdown` or similar.
- @diodon suggested the use of tabs to show code snippets for different languages. This can be done with `{.tabset}` in `markdown`, but this option does not work in GitHub. For what I have seen, it should work if used with `bookdown`.
- I realised that I do not have much information about how the API works. I was looking at the documentation, but seems unfinished. This section will need to be expanded later.
- The `README` is quite long now and I have not finished editing. Should we consider start developing the website instead. Otherwise I feel users may find this too cumbersome.

lidefi87 committed:
testing tabs in readme

lidefi87 committed:
Merge branch 'main' of https://github.com/lidefi87/rimrep-examples into main

lidefi87 committed:
checking tabs

lidefi87 committed:
adding figures used in readme update

lidefi87 committed:
adding python packages used in SELTMP notebook

lidefi87 committed:
reorganising readme

lidefi87 committed:
adding more snippets to readme

barbuz commented:
Regarding the different language tabs, I believe `{.tabset}` is a feature specific to R. There does not seem to be an equivalent in GitHub, but something that could work well enough is the [`<details>` tag ](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-collapsed-sections). 
We could have code examples initially hidden, and people will be able to click to see only those that interest them.

barbuz commented on README.md:
```diff
@@ -1,10 +1,166 @@
+There are two main ways to browse datasets available in the RIMReP DMS:  
+  
+1. Via the SpatioTemporal Asset Catalogs (STAC). STAC provides a common language to describe a range of geospatial information, so that data can be indexed and easily discovered. Our STAC catalogue is available at [https://stac.staging.reefdata.io/](https://stac.staging.reefdata.io/).  
+2. Via Pygeoapi, which provides API access to geospatial data that is compliant with OGC API standards. Our Pygeoapi service is available at [https://pygeoapi.staging.reefdata.io/](https://pygeoapi.staging.reefdata.io/).  
```
I think we should present STAC as the only way to search for datasets. We expect people to get the link to specific datasets on pygeoapi after looking them up on STAC, presenting them together like this can confuse people as to which one they are supposed to use.

diodon replied:
Agree. We must encourage the user to use STAC-browser

barbuz commented on README.md:
```diff
@@ -1,10 +1,166 @@
+### Searching for datasets via STAC
+The [STAC catalogue](https://stac.staging.reefdata.io/) is a web-based interface that allows users to search for datasets using a range of filters, such as dataset name, data provider, and date range. To search for datasets, you have the option of clicking on the **Search** button on the top right corner of the page, or you can use the search bar on the top left corner of the page. These two opti ...
+  
+![Screenshot of STAC catalogue home page showing the two search options mentioned in the previous paragraph](images/STAC_home.png)
```
```suggestion
![Screenshot of STAC catalogue home page showing the two search options mentioned in the previous paragraph](images/stac_home.png)

```

lidefi87 replied:
oops! I've fixed this now

barbuz reviewed (changes requested):
I've finally managed to read it, thanks for writing this!

Here are some suggestions I have:
- A table of contents near the beginning of the file could be useful, to let people jump to the section they are looking for
- We need examples collecting data from either S3 or pygeoapi, and a paragraph describing the differences between the two alternatives, I can help writing it (but maybe not very soon).
-  Same for managing data in Parquet or Zarr. The R section for this should note that for now you have to download the data as a NetCDF file from pygeoapi and show how to process it.

The documentation link for pygeoapi is indeed broken, I should try to fix it as soon as I have time, but for now don't worry about that part.

I think this could work well as a GitHub wiki page, but I also don't have permission to create one. @diodon can probably do it (see https://docs.github.com/en/communities/documenting-your-project-with-wikis/disabling-wikis)

lidefi87 committed:
checking tabs

lidefi87 committed:
correcting tabs

lidefi87 committed:
checking text formatting

lidefi87 committed:
edits suggested by Leo G

lidefi87 committed:
adding snippet for gridded data access

barbuz commented:
> I have not managed to make zarr work in R. I cannot seem to find a way to even search for what I need. It seems like arrow is good only for tabular data....at least for what I can see.

Yes, arrow is not made for tabular data, both Python and R need different libraries to manage raster data. For Python that would usually be `xarray`, for R I believe [stars](https://r-spatial.github.io/stars/) could be what we need; I tried making it work with Zarr data on s3 in the past and could not find a way to do it, that's why I think we'll likely need to download a portion of a dataset with pygeoapi and open the downloaded file with `stars`.

michaelamiller12 commented on README.md:
```diff
@@ -1,26 +1,238 @@
+## Searching for datasets in RIMReP DMS
+There are two main ways to browse datasets available in the RIMReP DMS:  
+  
+1. Via the SpatioTemporal Asset Catalogs (STAC). STAC provides a common language to describe a range of geospatial information, so that data can be indexed and easily discovered. Our STAC catalogue is available at [https://stac.staging.reefdata.io/](https://stac.staging.reefdata.io/).  
```
Might be better if the link is to the STAC Browser? https://stac.staging.reefdata.io/browser/

lidefi87 replied:
Thanks for that. I have updated this

michaelamiller12 commented:
It looks absolutely BEAUTIFUL @lidefi87 - I love it

lidefi87 commented:
> ...that's why I think we'll likely need to download a portion of a dataset with pygeoapi and open the downloaded file with `stars`.

I think this may be the way to go because I have been looking at just querying what is in the S3 bucket and cannot seem to find a way. I will play around with this option based on the examples in the other notebooks. Thanks @barbuz

lidefi87 committed:
updating STAC url

lidefi87 committed:
negrita

lidefi87 committed:
splitting readme into pages

lidefi87 committed:
sections as pages

lidefi87 committed:
moving folders

lidefi87 committed:
testing figure rendering

lidefi87 committed:
image relocated

lidefi87 committed:
Update rimrep_api_access.md

lidefi87 committed:
Update rimrep_api_access.md

lidefi87 committed:
dividing readme into multiple pages

lidefi87 committed:
accessing raster data via API call

lidefi87 committed:
updating name of R script called in notebook

lidefi87 committed:
renaming script because non-spatial functions are now included

lidefi87 committed:
updating names of scripts

lidefi87 committed:
function accessing DHW from API

lidefi87 committed:
updating libraries needed to run R notebooks

lidefi87 committed:
adding option for spatial boundaries in gridded data extraction from API

lidefi87 committed:
correcting mistake when creating raster from array

lidefi87 committed:
Adding notebook showing how to access gridded data from API

lidefi87 committed:
remove script showing use of connect_dms_dataset function. Replaced by markdown

lidefi87 commented:
This version has a function called `connect_dms_dataset` in the `useful_functions.R` script that allows users to connect to the API and extract data based using spatial and temporal boundaries. Currently, the function assumes that coordinates are called `lon` and `lat`. This needs to be updated to read the metadata for a collection to get the correct names for coordinates. There is now a new `Rmarkdown` notebook showing the use of this function.

This PR should address the following issues:
[#528](https://github.com/aodn/rimrep-dms/issues/528) - Accessing API from `R` example notebooks
[#531](https://github.com/aodn/rimrep-dms/issues/531) - Accessing gridded data in `R`
[#667](https://github.com/aodn/rimrep-dms/issues/667) - Improving `README`

@diodon, @barbuz, could you let me know if you are happy with this or if any changes should be applied?

lidefi87 committed:
adding R gridded data access snippet

diodon commented on README.md:
````diff
@@ -1,72 +1,244 @@
-  checking_libraries()
-```  
-The lines above will run a function that automatically checks if any `R` libraries used in this repository are not installed in your machine. If any libraries are missing, it will install them automatically. Bear in mind that these notebooks were developed in `R` version 4.3.1, so you may need to upgrade your `R` version if you encounter any problems during package installation.
+In simple terms, RIMReP DMS is a data portal that aims to be a "one-stop-shop" for all data related to the Great Barrier Reef World Heritage Area, which can be easily accessed by the Great Barrier Reef Marine Park Authority (GBRMPA) to support evidence-based management strategies. All datasets have a standard format regardless of their origin, which not only facilitates access to data, but also t ...
````
> but also their analysis as it removes the need to quality control individual datasets.  
This is not true. Replace with: 
but also their analysis as it removes the need to understand and parse different data formats

lidefi87 committed:
comments by Eduardo addressed

lidefi87 commented:
I have gone through Eduardo's comments and addressed them. I think this PR also addressed the issues below:

> [#528](https://github.com/aodn/rimrep-dms/issues/528) - Accessing API from `R` example notebooks
> [#531](https://github.com/aodn/rimrep-dms/issues/531) - Accessing gridded data in `R`
> [#667](https://github.com/aodn/rimrep-dms/issues/667) - Improving `README`

Do you think these issues are addressed by this PR? Should I mark them as complete?

barbuz reviewed (approved):
I think this looks good and the issues you mentioned can all be closed together with this.

As a small comment on the Python instructions, I think the fact that we show `pyarrow` to connect to a dataset and `dask-geopandas` to process it could be a bit confusing as in fact those are two possible alternatives. Maybe using `dask-geopandas` everywhere could be easier to follow for an inexperienced user?

lidefi87 commented:
I think `pyarrow` is nice because it lets the user inspect the data contained inside the dataset. While `dask-geopandas` needs you to provide the correct variable names to extract the data. I think it is worth keeping both of them, otherwise we will need to explain how we got the variable names when using `dask-geopandas` only
